apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -787,3 +812,106 @@ get_block_hash_optionally_by_hash_or_height(Req) ->
+          parse_filter_param(exclude_tx_types, Req)} of
+        {{error, unknown_type} = Err, _} ->
+            Err;
+        {_, {error, unknown_type} = Err} ->
+            Err;
+        {{ok, KeepTxTypes}, {ok, DropTxTypes}} ->
+            %% the logic bellow is a subject to some refactoring as soon as 
+            %% coinbase_tx become distinguishable and having disting tx hashes
+            %% current solution is a workaround
+            NonCoinbases = get_non_coinbase_txs_and_headers(KeepTxTypes,
+                                                            DropTxTypes,
+                                                            Account),
+            CoinbaseType = aec_coinbase_tx:type(),
+            CoinbaseRequested = lists:member(CoinbaseType, KeepTxTypes)
+                  orelse KeepTxTypes =:= [],
+            Coinbases =
+                case CoinbaseRequested andalso
+                    not lists:member(CoinbaseType, DropTxTypes) of
+                    true -> get_coinbase_txs_and_headers(Account);
+                    false -> []
+                end,
+            Res =
+              lists:sort(
+                  fun({HeaderA, _}, {HeaderB, _}) ->
+                      aec_headers:height(HeaderA) > aec_headers:height(HeaderB)
+                  end,
+                  NonCoinbases ++ Coinbases),
+            {ok, offset_and_limit(Req, Res)}
+      end.
+
+%% WORKAROUND
+%% Currently all coinbase txs are the same and thus having the same tx hash
+%% There is no point in fetching them via aec_db:dirty_transactions_by_account/2
+%% TODO: remove this workaround
+get_coinbase_txs_and_headers(Account) ->
+    {ok, SampleCoinbaseTx} = aec_coinbase_tx:new(#{account => Account}),
+    CoinbaseHash = aec_tx:hash_tx(SampleCoinbaseTx),
+    %% all coinbases share the same transaction hash and we can fetch them
+    lists:map(
+        fun({BlockHash, SignedCoinbaseTx}) ->
+            {ok, Header} = aec_conductor:get_header_by_hash(BlockHash),
+            {Header, SignedCoinbaseTx}
+        end,
+        aec_db:read_tx(CoinbaseHash)).
+
+%% WORKAROUND
+%% Currently all coinbase txs are the same and thus having the same tx hash
+%% The following function is the correct behavior with removing the commented
+%% line that filters out coinbase types
+%% TODO: remove this workaround
+get_non_coinbase_txs_and_headers(KeepTxTypes, DropTxTypes, Account) ->
+    CoinbaseType = aec_coinbase_tx:type(),
+    Filter =
+        fun(SignedTx) ->
+              Tx = aec_tx_sign:data(SignedTx),
+              Mod = aec_tx_dispatcher:handler(Tx),
+              TxType = Mod:type(),
+              Drop = lists:member(TxType, DropTxTypes)
+                  orelse TxType =:= CoinbaseType, %% later remove this
+              Keep = KeepTxTypes =:= []
+                  orelse lists:member(TxType, KeepTxTypes),
+              Keep andalso not Drop
+        end,
+    Txs = aec_db:dirty_transactions_by_account(Account, Filter),
```

Comment:
This will get all filtered transactions? 😱 
The pagination should probably be pushed inwards... Maybe not now, but later.